### DISCUSSION TITLE
e_sqlite3.dll copied to wrong location

### DISCUSSION THREAD

**Discussion by mbae23** [ericsink/SQLitePCL.raw#493] [Q&A]
Hello everybody.

We have a Xamarin Forms app which we want to build with GTK, with the final goal of running it under Linux. But for my question, we can stick to Windows.
Before I updated the SQLitePCL NuGet packages, the build copied e_sqlite3.dll to ...\bin\Debug\runtimes\win-x86 and it was found at startup. (For completeness, it was also copied to ...runtimes\win-arm and ...runtimes\win-x64.) The packages were at 2.0.7.

Then I updated, which means I removed sqlite-net-pcl and all its dependencies, added it again and updated all the packages it depends on to the newest version. Which is 2.1.0 and 1.1.14, respectively. Please see the .csproj file for details.
Now e_sqlite3.dll is copied to ...\bin\Debug\x86 and ...\bin\Debug\x64, and they are no longer found at startup.

Here is the exception thrown at startup. (Please ignore the funny mix of English and German.)

> System.TypeInitializationException: Der Typeninitialisierer für "SQLite.SQLiteConnection" hat eine Ausnahme verursacht. ---> System.Exception: Library e_sqlite3 not found
> plat: win
> suffix: DLL
> possibilities (2):
>    1) C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2.GTK\bin\Debug\runtimes\win-x86\native\e_sqlite3.dll
>    2) C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2.GTK\bin\Debug\e_sqlite3.dll
>win TryLoad: C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2.GTK\bin\Debug\runtimes\win-x86\native\e_sqlite3.dll
>thrown: System.ComponentModel.Win32Exception (0x80004005): Das angegebene Modul wurde nicht gefunden
>   bei SQLitePCL.NativeLibrary.TryLoad(String name, Loader plat, Action1 log, IntPtr& h)
> win TryLoad: C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2.GTK\bin\Debug\e_sqlite3.dll
> thrown: System.ComponentModel.Win32Exception (0x80004005): Das angegebene Modul wurde nicht gefunden
>    bei SQLitePCL.NativeLibrary.TryLoad(String name, Loader plat, Action1 log, IntPtr& h)
> NOT FOUND
>
>   bei SQLitePCL.NativeLibrary.Load(String libraryName, Assembly assy, Int32 flags)
>   bei SQLitePCL.Batteries_V2.MakeDynamic(String name, Int32 flags)
>   bei SQLitePCL.Batteries_V2.DoDynamic_cdecl(String name, Int32 flags)
>   bei SQLitePCL.Batteries_V2.Init()
>   bei SQLite.SQLiteConnection..cctor()
>   --- Ende der internen Ausnahmestapelüberwachung ---
>   bei SQLite.SQLiteConnection..ctor(String databasePath, SQLiteOpenFlags openFlags, Boolean storeDateTimeAsTicks)
>   bei Digi.App2.GTK.PlatformServices.SQLiteLinux.Connect() in C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2.GTK\PlatformServices\SQLiteLinux.cs:Zeile 27.
>   bei Digi.App2.GTK.PlatformServices.SQLiteLinux.OpenDatabase(Logger logger) in C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2.GTK\PlatformServices\SQLiteLinux.cs:Zeile 39.
>   bei Digi.App2.App..ctor() in C:\Entwicklung\DIGI6\work-app2\Source\Digi.App2\Digi.App2\App.cs:Zeile 294.

I need your help here. Am I missing something, like another NuGet package or config change/tweak or is this a bug?
P.S.: Sure, the workaround is to copy the DLL to one of the search locations and everything works. But the build should work.
P.P.S.: If you want me to list the relevant packages from the .csproj here then please let me know. Wasn't sure about the best way to share this file.

[Digi.App2.GTK.csproj.txt](https://github.com/ericsink/SQLitePCL.raw/files/8815887/Digi.App2.GTK.csproj.txt)

**Comment by ericsink** [comment]
Looks like you're using `packages.config`?  If so, then this is related to #492.

**Reply by mbae23** [reply]
Hello Eric,

thanks for the super fast reply. Yes the project is still using `packages.config`, so #492 seems to be the problem we're having. Not sure when we'll be able to switch to the new project format, we're in the same position as the author of the bug.

Best regards,
Markus
